tests/unit/amo/components/TestRating.js

```diff
@@ -69,411 +95,333 @@ describe(__filename, () => {
-    });
-
-    it("passes several props to the IconStar component when it's readOnly", () => {
-      const root = render({ readOnly: true });
-
-      const star = root.find(IconStar).at(0);
+      render({ readOnly: false });
+      const firstRating = screen.getByClassName('Rating-rating-1');
 
-      expect(star).toHaveProp('selected');
-      expect(star).toHaveProp('half');
-      expect(star).toHaveProp('yellow');
-      expect(star).toHaveProp('readOnly', true);
+      // This is the result of "selected".
+      expect(within(firstRating).getByTagName('path')).toHaveAttribute(
+        'd',
+        getSvgPath(OPEN_STYLE),
+      );
+      expect(within(firstRating).getByTagName('g')).toHaveAttribute(
+        'fill',
+        photon.YELLOW_50,
+      );
     });
-  });
 
-  describe('rating counts', () => {
-    it('renders the average rating', () => {
-      const rating = 3.5;
-      const root = render({ rating, readOnly: true });
-
-      expect(root).toHaveProp('title', `Rated ${rating} out of 5`);
-    });
+    it("passes expected props to the IconStar component when it's readOnly", () => {
+      render({ readOnly: true });
+      const firstRating = screen.getByClassName('Rating-rating-1');
 
-    it('localizes average rating', () => {
-      const i18n = fakeI18n({ lang: 'de' });
-      expect(render({ rating: 3.5, i18n, readOnly: true }).text()).toContain(
-        '3,5',
+      // This is the result of the default props from readOnly.
+      expect(within(firstRating).getByTagName('path')).toHaveAttribute(
+        'd',
+        getSvgPath(DIM_CLOSED_STYLE),
       );
     });
+  });
 
-    it('renders empty ratings', () => {
-      const root = renderWithEmptyRating({ readOnly: true });
-
-      expect(root).toHaveProp('title', 'There are no ratings yet');
-    });
+  it('localizes the rating title', () => {
+    const i18n = fakeI18n({ lang: 'de' });
+    render({ rating: 3.5, i18n, readOnly: true });
+    expect(screen.getAllByTitle('Rated 3,5 out of 5')).toHaveLength(6);
   });
 
   describe('loading state', () => {
     it('enters a loading state without a rating', () => {
-      const root = render({ rating: undefined });
+      render({ rating: undefined });
 
-      expect(root).toHaveClassName('Rating--loading');
+      expect(screen.getByClassName('Rating')).toHaveClass('Rating--loading');
     });
 
     it('exits the loading state with a rating value', () => {
-      const root = render({ rating: 4 });
+      render({ rating: 4 });
 
-      expect(root).not.toHaveClassName('Rating--loading');
+      expect(screen.getByClassName('Rating')).not.toHaveClass(
+        'Rating--loading',
+      );
     });
 
     it('exits the loading state with an empty rating', () => {
-      const root = renderWithEmptyRating();
+      renderWithEmptyRating();
 
-      expect(root).not.toHaveClassName('Rating--loading');
+      expect(screen.getByClassName('Rating')).not.toHaveClass(
+        'Rating--loading',
+      );
```

Comment:
In https://github.com/mozilla/addons-frontend/pull/11336/files we've been doing tests like: 
`expect(within(screen.getByClassName('UserProfile-name')).getByRole('alert')).toBeInTheDocument();` 
to check for loading state